pyaedt/application/Design.py

```diff
@@ -4020,32 +4021,61 @@ def set_temporary_directory(self, temp_dir_path):
         return True
 
     @pyaedt_function_handler()
+    def _manipulate_design_settings_inputs(self, k, v):
+        return v
+
+    @property
     def design_settings(self):
-        """Get design settings for the current AEDT app.
+        return DesignSettings(self, self._manipulate_design_settings_inputs)
 
-        Returns
-        -------
-        dict
-            Dictionary of valid design settings.
 
-        References
-        ----------
+class DesignSettings:
+    """Get design settings for the current AEDT app.
 
-        >>> oDesign.GetChildObject("Design Settings")
-        """
+    References
+    ----------
+
+    >>> oDesign.GetChildObject("Design Settings")
+    """
+
+    def __init__(self, app, manipulate_inputs):
+        self._app = app
+        self._manipulate_inputs = manipulate_inputs
         try:
-            design_settings = self._odesign.GetChildObject("Design Settings")
-        except Exception:  # pragma: no cover
-            self.logger.error("Failed to retrieve design settings.")
-            return False
+            self.design_settings = self._app.odesign.GetChildObject("Design Settings")
+        except GrpcApiError:  # pragma: no cover
+            self._app.logger.error("Failed to retrieve design settings.")
+            self.design_settings = None
 
-        prop_name_list = design_settings.GetPropNames()
-        design_settings_dict = {}
-        for prop in prop_name_list:
-            try:
-                design_settings_dict[prop] = design_settings.GetPropValue(prop)
-            except Exception:  # pragma: no cover
-                self.logger.warning('Could not retrieve "{}" property value in design settings.'.format(prop))
-                design_settings_dict[prop] = None
+    @property
+    def available_properties(self):
+        return [prop for prop in self.design_settings.GetPropNames() if not prop.endswith("/Choices")]
+
+    def __repr__(self):
+        lines = ["{"]
+        for prop in self.available_properties:
+            lines.append("\t{}: {}".format(prop, self.design_settings.GetPropValue(prop)))
+        lines.append("}")
+        return "\n".join(lines)
+
+    def __setitem__(self, key, value):
+        if key in self.available_properties:
+            value = self._manipulate_inputs(key, value)
+            key_choices = "{}/Choices".format(key)
+            if key_choices in self.design_settings.GetPropNames():
+                value_choices = self.design_settings.GetPropValue(key_choices)
+                if value not in value_choices:
+                    self._app.logger.error(
+                        "{} is not a valid choice. Possible choices are: {}".format(value, ", ".join(value_choices))
+                    )
+                    return False
+            self.design_settings.SetPropValue(key, value)
+        else:
+            self._app.logger.error("{} property not available in design settings.".format(key))
```

Comment:
```suggestion
            self._app.logger.error("{} property is not available in design settings.".format(key))
```
Message strrings should be complete sentences. Added the verb.